### PR TITLE
[RELEASE-1.17] Adding the post intsall job for 1.17/0.23

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

as per title.

We do already build the image for the job for each release (e.g. `registry.ci.openshift.org/openshift/knative-v0.23.0:knative-eventing-storage-version-migration`) and that is already part of the `ClusterServiceVersion` manifest we ship 

/hold

### CODE REVIEW

**Comment by matzew** [comment]
```
failed to load configuration: got unexpected http 504 status code from configresolver: <html><body><h1>504 Gateway Time-out</h1> The server didn't respond in time. </body></html>
```

/retest

File: openshift-knative-operator/cmd/operator/kodata/knative-eventing/0.23.0/6-eventing-post-install.yaml [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+      annotations:
+        sidecar.istio.io/inject: "false"
+    spec:
+      serviceAccountName: knative-eventing-post-install-job
+      restartPolicy: OnFailure
+      containers:
+        - name: migrate
+          image: TO_BE_REPLACED
+          args:
+            - "pingsources.sources.knative.dev"
```
[the remark this answers is not in the document]
Comment (matzew) [reply openshift-knative-operator/cmd/operator/kodata/knative-eventing/0.23.0/6-eventing-post-install.yaml:186]: we might do that in a separate PR - perhaps even later (see upstream discussions)

This job here was originally done for the pingsource only - so, I wanna keep it that way, and not mix stuff - if not really really needed

**Comment by lberk** [comment]
infra fail,
/test 4.6-upgrade-tests-aws-ocp-46

File: openshift-knative-operator/hack/update-manifests.sh [hunk 2]
```diff
@@ -117,3 +117,6 @@ git apply "$root/openshift-knative-operator/hack/007-networkpolicy-mesh.patch"
 # Add the old versions to eventing CRDs.
 # TODO: Remove after the version migrator is set to always run in eventing
 git apply  "$root/openshift-knative-operator/hack/008-eventing-crds-old-versions.patch"
+
+# Correct names of the batch jobs that we forgot on 1.16
+git apply "$root/openshift-knative-operator/hack/008-fix-post-install-names.patch"
```
Comment (devguyio) [inline openshift-knative-operator/hack/update-manifests.sh:122]: is there a reason you're using `008` prefix that's already used for the previous patch?

**Review by devguyio — COMMENTED** [review]
@matzew for this webhook to work, we'll need to serve for PingSource v1alpha2 and v1beta1 ( i.e `served: true` )

File: openshift-knative-operator/hack/update-manifests.sh (hunk 2 above)
Comment (matzew) [reply openshift-knative-operator/hack/update-manifests.sh:122]: nah, I can change that - did not see the other 008 LOL

**Comment by matzew** [comment]
> @matzew for this webhook to work, we'll need to serve for PingSource v1alpha2 and v1beta1 ( i.e served: true )

@devguyio ok - but let's do that on a separate PR, for the CRDs

File: openshift-knative-operator/hack/update-manifests.sh (hunk 2 above)
Comment (matzew) [reply openshift-knative-operator/hack/update-manifests.sh:122]: will change the file name

**Comment by lberk** [comment]
/lgtm
/unhold
unholding due to openshift-knative/serverless-operator#1186

**Pull request merged**